app/models/post.rb

```diff
@@ -1,2 +1,13 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def update_posts_counter
+    user.update(post_counter: user.posts.count)
+  end
```

Comment:
- [ Optional ] I think it would be nice if you make use of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_posts_counter` method every time a new post is saved to the database, then make the `update_post_counter` method private. Please see the example below for a suggestion on how to achieve this.

## Example code
```ruby
class Post < ApplicationRecord
  belongs_to :author, class_name: 'User'
  has_many :comments
  has_many :likes

  after_save :update_posts_counter

  def recent_comments
    comments.order(created_at: :desc).limit(5)
  end
  
  private

  def update_posts_counter
    user.update(post_counter: user.posts.count)
  end
end
```